src/cleanlab_codex/codex_tool.py

```diff
@@ -126,7 +126,7 @@ def to_smolagents_tool(self) -> Any:
         try:
             from cleanlab_codex.utils.smolagents import CodexTool as SmolagentsCodexTool
         except ImportError as e:
-            raise MissingDependencyError("smolagents", "https://github.com/huggingface/smolagents") from e
+            raise MissingDependencyError(e.name or "smolagents", "https://github.com/huggingface/smolagents") from e
```

Comment:
Doesn't this supply the URL as the `package_name` argument to `MissingDependencyError.__init__`, rather than as the `package_url` argument?